api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -648,17 +649,17 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
       String fromWorkspaceNamespace,
       String fromWorkspaceId,
       String fromNotebookName,
-      CopyNotebookRequest copyNotebookRequest) {
+      CopyRequest copyRequest) {
     FileDetail fileDetail;
     try {
       fileDetail =
           notebooksService.copyNotebook(
               fromWorkspaceNamespace,
               fromWorkspaceId,
-              fromNotebookName,
-              copyNotebookRequest.getToWorkspaceNamespace(),
-              copyNotebookRequest.getToWorkspaceName(),
-              copyNotebookRequest.getNewName());
+              fromNotebookName.concat(NOTEBOOK_EXTENSION),
+              copyRequest.getToWorkspaceNamespace(),
+              copyRequest.getToWorkspaceName(),
+              copyRequest.getNewName().concat(NOTEBOOK_EXTENSION));
     } catch (BlobAlreadyExistsException e) {
       throw new BadRequestException("File already exists at copy destination");
```

Comment:
Instead of a 400, we could do a 409 here, since there's a conflict with an existing resource.

Reply:
👍